crates/primitives/src/identity.rs

```diff
@@ -1,7 +1,19 @@
+use borsh::{BorshDeserialize, BorshSerialize};
 use serde::{Deserialize, Serialize};
 
 use crate::context::ContextId;
 
+#[derive(Eq, Clone, Debug, PartialEq, BorshSerialize, BorshDeserialize)]
+pub struct ContextIdentity {
+    pub public_key: [u8; 32],
```

Comment:
you can define a new type here to represent the public key, I suggest following the pattern used for `ContextId`

Reply:
I've introduced `KeyPair` and `PublicKey` to be more appropriate here, with `ContextIdentity` now representing semantic usage elsewhere.